src/fs/rock/RockSwapDir.cc

```diff
@@ -54,25 +54,30 @@ Rock::SwapDir::~SwapDir()
 
 // called when Squid core needs a StoreEntry with a given key
 StoreEntry *
-Rock::SwapDir::get(const cache_key *key)
+Rock::SwapDir::get(const Store::CacheKey &cacheKey)
 {
     if (!map || !theFile || !theFile->canRead())
         return NULL;
 
     sfileno filen;
-    const Ipc::StoreMapAnchor *const slot = map->openForReading(key, filen);
+    const Ipc::StoreMapAnchor *const slot = map->openForReading(cacheKey.key, filen);
     if (!slot)
         return NULL;
 
     // create a brand new store entry and initialize it with stored basics
     StoreEntry *e = new StoreEntry();
+    e->createMemObject();
     anchorEntry(*e, filen, *slot);
 
-    e->hashInsert(key);
-    trackReferences(*e);
-
-    return e;
-    // the disk entry remains open for reading, protected from modifications
+    e->hashInsert(cacheKey.key);
+    if (e->preparePublicEntry(cacheKey, true)) {
```

Comment:
SwapDir::get() reads from the cache. Per our [earlier agreement](https://github.com/squid-cache/squid/pull/46#discussion_r141682972), readers do not need Transient entries (only writers must create them). This concern is similar to the StoreMap::get() concern discussed in another review comment.